From 954a49ec0a4dc64fd9e513abe7a737d956b337ca Mon Sep 17 00:00:00 2001
From: Yi Zhao <yi.zhao@windriver.com>
Date: Tue, 9 Feb 2021 17:50:24 +0800
Subject: [PATCH] policy/modules/system/systemd: allow systemd-generators to
 get the attributes of tmpfs and cgroup

* Allow systemd-generators to get the attributes of a tmpfs
* Allow systemd-generators to get the attributes of cgroup filesystems

Fixes:
systemd[95]: /lib/systemd/system-generators/systemd-fstab-generator failed with exit status 1.

avc: denied { getattr } for pid=102 comm="systemd-run-gen" name="/"
dev="tmpfs" ino=11208 scontext=system_u:system_r:systemd_generator_t
tcontext=system_u:object_r:tmpfs_t tclass=filesystem permissive=0

avc: denied { getattr } for pid=98 comm="systemd-getty-g" name="/"
dev="tmpfs" ino=11208 scontext=system_u:system_r:systemd_generator_t
tcontext=system_u:object_r:tmpfs_t tclass=filesystem permissive=0

avc: denied { getattr } for pid=104 comm="systemd-sysv-ge" name="/"
dev="tmpfs" ino=11208 scontext=system_u:system_r:systemd_generator_t
tcontext=system_u:object_r:tmpfs_t tclass=filesystem permissive=0

avc:  denied  { getattr } for pid=97 comm="systemd-fstab-g" name="/"
dev="tmpfs" ino=11208 scontext=system_u:system_r:systemd_generator_t
tcontext=system_u:object_r:tmpfs_t tclass=filesystem permissive=0

avc:  denied  { getattr } for  pid=102 comm="systemd-run-gen" name="/"
dev="cgroup" ino=1 scontext=system_u:system_r:systemd_generator_t
tcontext=system_u:object_r:cgroup_t tclass=filesystem permissive=0

avc:  denied  { getattr } for  pid=100 comm="systemd-hiberna" name="/"
dev="cgroup" ino=1 scontext=system_u:system_r:systemd_generator_t
tcontext=system_u:object_r:cgroup_t tclass=filesystem permissive=0

avc:  denied  { getattr } for  pid=99 comm="systemd-gpt-aut" name="/"
dev="cgroup2" ino=1 scontext=system_u:system_r:systemd_generator_t
tcontext=system_u:object_r:cgroup_t tclass=filesystem permissive=0

avc:  denied  { getattr } for  pid=97 comm="systemd-fstab-g"
path="/var/volatile" dev="vda" ino=37131
scontext=system_u:system_r:systemd_generator_t
tcontext=system_u:object_r:unlabeled_t tclass=dir permissive=0

Upstream-Status: Inappropriate [embedded specific]

Signed-off-by: Yi Zhao <yi.zhao@windriver.com>
---
 policy/modules/system/systemd.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/policy/modules/system/systemd.te b/policy/modules/system/systemd.te
index 2d9d7d331..c1111198d 100644
--- a/policy/modules/system/systemd.te
+++ b/policy/modules/system/systemd.te
@@ -431,6 +431,9 @@ files_list_usr(systemd_generator_t)
 
 fs_list_efivars(systemd_generator_t)
 fs_getattr_xattr_fs(systemd_generator_t)
+fs_getattr_tmpfs(systemd_generator_t)
+fs_getattr_cgroup(systemd_generator_t)
+kernel_getattr_unlabeled_dirs(systemd_generator_t)
 
 init_create_runtime_files(systemd_generator_t)
 init_manage_runtime_dirs(systemd_generator_t)
-- 
2.17.1

